containers-contrib-readonly/src/Map/Contrib/Bench.hs

```diff
@@ -0,0 +1,39 @@
+{-# LANGUAGE BangPatterns #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE NoImplicitPrelude #-}
+{-# LANGUAGE TypeFamilies #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE TypeApplications #-}
+
+module Map.Contrib.Bench where
+
+import Map
+import Gauge.Main (Benchmark, bench, defaultMain, whnf)
+import Relude hiding (fromList, Map)
+
+benchmark
+  :: forall k. (Enum k, Bounded k, Key k, NFData (Map k Int))
+  => String
+  -> Proxy k
```

Comment:
Instead of passing `Proxy k` here you can just not pass it and enable `AllowAmbiguousTypes` language extension. So instead of passing:

```haskell
benchmark "some label" (Proxy @T)
```

you can call this function just as:

```haskell
benchmark @T "some label"
```

`-XTypeApplications` FTW 👍

Reply:
Oh, I chose not to enable the `AllowAmbiguousTypes` extension but in this case it might be okay 😅

Reply:
Despite the name, `AllowAmbiguousTypes` is actually a completely safe extension 👌 You only can use it with `TypeApplications` but `TypeApplications` is also safe.